[MTIA] Enable deserialization for FP8 checkpoint loading

PatriceVignola opened the pull request:
Summary: It looks like loading FP8 checkpoints goes through that path which wasn't enabled for MTIA beforehand, whereas loading BF16 checkpoints didn't.

Differential Revision: D82997140

PatriceVignola commented:
@pytorchbot label "topic: not user facing"

mikaylagawarecki reviewed (commented):
Are there tests that exercise this change?

PatriceVignola committed:
[MTIA] Enable deserialization for FP8 checkpoint loading (#163559)

Summary:

It looks like loading FP8 checkpoints goes through that path which wasn't enabled for MTIA beforehand, whereas loading BF16 checkpoints didn't.

Test Plan: See the usual commands.

Reviewed By: mikaylagawarecki, brad-mengchi

Differential Revision: D82997140
